application/controllers/admin/DataEntry.php

```diff
@@ -1884,20 +1884,11 @@ public function insert()
                     }
                     // query for updating tokens uses left
                     $aToken = Token::model($surveyid)->findByAttributes(['token' => $_POST['token']]);
+                    $aToken->usesleft = ((int) $aToken->usesleft) - 1;
                     if (isTokenCompletedDatestamped($thissurvey)) {
-                        if ($aToken->usesleft <= 1) {
```

Comment:
keep this condition,  add "$aToken->completed = $submitdate;" in it

same for the other condition but with "$aToken->completed = 'Y';"
then it will be fine

Reply:
just had another look and need to make another adjustment, it can only be substracted after the condition was checked, so after the condition you need to add "$aToken->usesleft = ((int) $aToken->usesleft) - 1;" again